models/marts/core/int_all_columns.sql

```diff
@@ -0,0 +1,59 @@
+-- one row for each column in a node or source
+
+with
+
+stg_columns as (
+
+    select
+        node_unique_id,
+        name,
+        description,
+        data_type,
+        quote
+
+    from {{ ref('stg_columns') }}
+
+),
+
+stg_nodes as (
+
+    select
+        unique_id,
+        name,
+        resource_type,
+        file_path,
+        is_enabled,
+        database,
+        schema,
+        package_name,
+        alias
+
+    from {{ ref('stg_nodes') }}
+
+),
+
+final as (
+
+    select
+        stg_nodes.unique_id as node_unique_id,
+        stg_nodes.name as node_name,
+        stg_nodes.resource_type as node_resource_type,
+        stg_nodes.file_path as node_file_path,
+        stg_nodes.is_enabled as node_is_enabled,
+        stg_nodes.database as node_database,
+        stg_nodes.schema as node_schema,
+        stg_nodes.package_name as node_package_name,
+        stg_nodes.alias as node_alias,
+        stg_columns.name as name,
+        stg_columns.description as description,
+        stg_columns.data_type as data_type,
+        stg_columns.quote as quote
+
+    from stg_columns
+
+    right join stg_nodes
+        on stg_nodes.unique_id = stg_columns.node_unique_id
```

Comment:
2 questions/points 😄 

- sorry to be picky, but can we use a `left join`? In our code we try to only use `left` or `inner`/`outer` joins, and never a `right`. This is just to make the mental models of joins easier to understand for someone seeing the code
- why would we want a `left`/`'right`, instead of a `inner` join? Do you want to use this table to see which table doesn't have any column configured in the YAML?

Reply:
Yes, this was my idea to not drop any nodes. But you are right, this is already intermediate layer, thus we can be dropping some data in order to expose higher quality data to the user.

I've changed the join to be inner.